app/Exceptions/Handler.php

```diff
@@ -48,6 +49,40 @@ public function report(Throwable $e)
         parent::report($e);
     }
 
+    protected function buildExceptionContext(Throwable $e)
+    {
+        $context = parent::buildExceptionContext($e);
+
+        $request = request();
+        if ($request) {
+            $context['url'] = $request->url();
+
+            // never log raw passwords
+            $params = Arr::except(request()->all(), $this->dontFlash);
+
+            // extract the user and token parameters for API calls
+            if (str_ends_with($context['url'], 'dorequest.php')) {
+                unset($params['u']);
+                unset($params['t']);
+            }
+            elseif (str_contains($context['url'], '/API/')) {
+                unset($params['z']);
+                unset($params['y']);
+            }
+
+            // truncate long parameters
+            foreach ($params as $k => $p) {
+                if (strlen($p) > 20)
+                    $params[$k] = substr($p, 0, 15) + "...";
```

Comment:
```suggestion
                    $params[$k] = substr($p, 0, 15) . "...";
```

nit: Does the `+` operator work for concatenation with a string here?

Reply:
Nope, good call.